public/pages/Correlations/containers/CorrelationsContainer.tsx

```diff
@@ -595,33 +999,49 @@ export class Correlations extends React.Component<CorrelationsProps, Correlation
                     Reset filters
                   </EuiSmallButtonEmpty>
                 </EuiFlexItem>
-              </EuiFlexGroup>
-              <EuiSpacer />
-              <EuiFlexGroup
-                wrap={true}
-                gutterSize="m"
-                justifyContent="flexStart"
-                alignItems="center"
-              >
                 <EuiFlexItem grow={false}>
-                  <EuiText size="s">
-                    <strong>Severity:</strong>
-                  </EuiText>
+                  <EuiButtonGroup
+                    legend="View type"
+                    options={[
+                      {
+                        id: 'table',
+                        label: 'Table',
+                        iconType: 'tableOfContents',
+                      },
+                      {
+                        id: 'graph',
+                        label: 'Graph',
+                        iconType: 'visNetwork',
+                      },
+                    ]}
+                    idSelected={this.state.isGraphView ? 'graph' : 'table'}
+                    onChange={(id) => this.setState({ isGraphView: id === 'graph' })}
+                    buttonSize="s"
+                  />
                 </EuiFlexItem>
-                {ruleSeverity.map((sev, idx) => (
-                  <EuiFlexItem grow={false} key={idx}>
-                    <EuiText size="s">
-                      <EuiIcon type="dot" color={sev.color.background} /> {sev.value}
-                    </EuiText>
-                  </EuiFlexItem>
-                ))}
               </EuiFlexGroup>
-
               <EuiSpacer />
-              {this.renderCorrelationsGraph(this.state.loadingGraphData)}
+              {this.state.isGraphView ? (
+                this.renderCorrelationsGraph(this.state.loadingGraphData)
+              ) : (
+                <>
+                  {this.renderCorrelatedFindingsChart()}
+                  {this.renderCorrelationsTable(this.state.loadingTableData)}
```

Comment:
Can these be their own components with container maintaining shared state? This file is still too big